2021/day15/day15.java

```diff
@@ -0,0 +1,146 @@
+import java.io.File;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.PriorityQueue;
+import java.util.Scanner;
+import java.lang.Integer;
+
+public class day15 {
+    public static class Position implements Comparable<Position> {
+        int x;
+        int y;
+        int risk;
+
+        Position(int vertical, int horizontal, int currentRisk) {
+            x = vertical;
+            y = horizontal;
+            risk = currentRisk;
+        }
+
+        @Override
+        public int compareTo(Position other) {
+            return Integer.compare(this.risk, other.risk);
+        }
+    }
+
+    public static ArrayList<ArrayList<Integer>> readInput(String filename) {
+        ArrayList<ArrayList<Integer>> input = new ArrayList<>();
+        Scanner sc = null;
+        try {
+            sc = new Scanner(new File(filename));
+            while (sc.hasNextLine()) {
+                ArrayList<Integer> line = new ArrayList<>();
+                sc.useDelimiter("");
+                while (sc.hasNextInt()) {
+                    line.add(sc.nextInt());
+                }
+                input.add(line);
+                if (sc.hasNextLine())
+                    sc.nextLine();
+            }
+        } catch (Exception e) {
+            System.out.println("Error handling input " + e);
+        } finally {
+            sc.close();
+        }
+        return input;
+    }
+
+    public static ArrayList<ArrayList<Integer>> modifyInput
```

Comment:
Right, simulating the full board is one way of approaching the problem. However this is in a way unnecessary. If you wanted to be space-efficient, you could write a method that would just retrieve a risk given `mapOfRisks`, `x`, and `y`. Something like

```pseudocode
int getRisk(mapOfRisks, x, y) {
  tmpRisk = mapOfRisks[y % height][x % width] + (x // width) + (y // height)
  return tmpRisk < 10 ? tmpRisk : ((tmpRisk % 10) + 1)
}
```